### PR TITLE
Add NUM_WORKERS variable to run_infer.sh scripts for configurable woker settings

### PR DESCRIPTION
**Pull request by neubig**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**

Fixes https://github.com/OpenDevin/OpenDevin/issues/2567

**Give a brief summary of what the PR does, explaining any non-trivial design decisions**

This makes NUM_WORKERS configurable in all evaluation scripts, and defaults it to 1.

Authored by OpenDevin.

### CODE REVIEW

File: evaluation/webarena/scripts/run_infer.sh [hunk 1]
```diff
@@ -35,7 +40,7 @@ COMMAND="poetry run python evaluation/webarena/run_infer.py \
   --llm-config $MODEL_CONFIG \
   --max-iterations 15 \
   --max-chars 10000000 \
-  --eval-note $EVAL_NOTE"
```
Comment (enyst) [inline evaluation/webarena/scripts/run_infer.sh:38]: Are the removals of EVAL_NOTE intended?

**Review by enyst — APPROVED** [review]

File: evaluation/webarena/scripts/run_infer.sh [hunk 2]
```diff
@@ -35,7 +40,7 @@ COMMAND="poetry run python evaluation/webarena/run_infer.py \
...
+  --eval-num-workers $NUM_WORKERS \
```
Comment (neubig) [inline evaluation/webarena/scripts/run_infer.sh:43]:
```suggestion
--eval-num-workers $NUM_WORKERS \
--eval-note $EVAL_NOTE"
```

File: evaluation/webarena/scripts/run_infer.sh (hunk 1 above)
Comment (neubig) [reply evaluation/webarena/scripts/run_infer.sh:38]: No, it was a mistake. Thanks for the catch!

**Comment by PierrunoYT** [comment]
@neubig Did you create this with OpenDevin?

**Comment by neubig** [comment]
Yes.

**Pull request merged**